### PR TITLE
Document tabs & add icon slot name

### LINKED ISSUE #586: Document tab bar & tab

### PR DESCRIPTION
**Pull request by e111077**
fixes #586

Documented tab and tab-bar as well as change the default `slot` to be `slot[name="icon"]`

Reasoning: it would break the following use case:

```html
<mwc-tab
  icon="camera">
</mwc-tab>
```

as the empty text node would override the default slot's contents.

Will require internal migration.

### CODE REVIEW

**Comment by e111077** [comment]
Running copybara to get ahead on internal fixes.

File: packages/tab-bar/README.md [hunk 1]
```diff
@@ -1,64 +1,278 @@
...
+</mwc-tab-bar>
+```
+
+## API
+
+### Slots
+
+| Name              |	Description
+| ----------------- | -------------
+| _default_         |	[`mwc-tab`](https://github.com/material-components/material-components-web-components/tree/master/packages/tab) elements to display.
```
Comment (aomarks) [inline packages/tab-bar/README.md]: s/default/icon

Comment (aomarks) [reply packages/tab-bar/README.md]: ping (new PR?)

Comment (e111077) [reply packages/tab-bar/README.md]: oh no, that's right, you're thinking of `packages/tab/README.md` not `tab-bar`

Comment (e111077) [reply packages/tab-bar/README.md]: sorry, I thought this comment was on the old PR and thought you were commenting on `tab`

**Pull request merged**